rclpy/test/test_executor.py

```diff
@@ -302,6 +302,47 @@ def test_executor_add_node(self):
         assert not executor.add_node(self.node)
         assert id(executor) == id(self.node.executor)
 
+    def test_executor_spin_until_future_complete(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.1, timer_callback)
+
+        from rclpy.task import Future
+
+        # Timeout
+        future = Future()
+        self.assertFalse(future.done())
+        start = time.monotonic()
+        executor.spin_until_future_complete(future=future, timeout_sec=1.0)
```

Comment:
To help cut down on test time, there shouldn't be an issue with a timeout value of 0.1s.

Reply:
Fixed